server/odc-common/src/main/java/com/oceanbase/odc/common/security/OdcBigDecimalChecker.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.common.security;
+
+import java.util.regex.Pattern;
+
+import com.oceanbase.odc.common.util.StringUtils;
+
+/**
+ * @author gaoda.xy
+ * @date 2024/4/22 15:31
+ */
+public class OdcBigDecimalChecker {
+
+    private static final Pattern DIGITS_PATTERN = Pattern.compile("[+-]?[0-9.]+");
+
+    private static final int MAX_EXPONENT_LENGTH = 4;
```

Comment:
is it better that this value can be configurable~

Reply:
There is a public method that can specify this value. 
For ODC, 4 (meansing 1e9999) is enough for almost all user usage, and is low cost for CPU processing.